src/mavsdk/plugins/camera/camera_definition.cpp

```diff
@@ -110,18 +110,21 @@ bool CameraDefinition::parse_xml()
             return false;
         }
 
-        const char* type_str = e_parameter->Attribute("type");
-        if (!type_str) {
+        const char* type_str_res = e_parameter->Attribute("type");
+        // The check for null pointer is required before the creation of std::string.
+        if (!type_str_res) {
             LogErr() << "type attribute missing for " << param_name;
             return false;
         }
 
-        if (std::string(type_str) == "string") {
+        // Convert to std::string only once.
```

Comment:
```suggestion
```

Reply:
Removed in [12a8070](https://github.com/mavlink/MAVSDK/pull/2056/commits/12a807014c365b61db423a6ff6ea94348167e060)